server/src/main/java/org/elasticsearch/tasks/Task.java

```diff
@@ -39,28 +47,26 @@ public class Task implements Traceable {
      */
     public static final String TRACE_PARENT_HTTP_HEADER = "traceparent";
 
+    public static final String TRACE_STATE = "tracestate";
+
     /**
-     * A request header that indicates the origin of the request from Elastic stack. The value will stored in ThreadContext
-     * and emitted to ES logs
+     * Parsed part of traceparent. It is stored in thread context and emitted in logs.
+     * Has to be declared as a header copied over for tasks.
      */
-    public static final String X_ELASTIC_PRODUCT_ORIGIN_HTTP_HEADER = "X-elastic-product-origin";
+    public static final String TRACE_ID = "trace.id";
 
-    public static final String TRACE_STATE = "tracestate";
+    public static final String TRACE_START_TIME = "trace.starttime";
 
     /**
      * Used internally to pass the apm trace context between the nodes
      */
     public static final String APM_TRACE_CONTEXT = "apm.local.context";
 
-    /**
-     * Parsed part of traceparent. It is stored in thread context and emitted in logs.
-     * Has to be declared as a header copied over for tasks.
-     */
-    public static final String TRACE_ID = "trace.id";
+    public static final String PARENT_TRACE_PARENT_HEADER = "parent_" + Task.TRACE_PARENT_HTTP_HEADER;
 
-    public static final String TRACE_START_TIME = "trace.starttime";
-    public static final String TRACE_PARENT = "traceparent";
-    public static final String X_ELASTIC_PROJECT_ID_HTTP_HEADER = "X-Elastic-Project-Id";
+    public static final String PARENT_TRACE_STATE = "parent_" + Task.TRACE_STATE;
+
+    public static final String PARENT_APM_TRACE_CONTEXT = "parent_" + Task.APM_TRACE_CONTEXT;
```

Comment:
I like the cleanup but it does rather complicate this change, and the way the names all change makes it hard to be confident in a review. Could you extract this more cosmetic stuff out into a PR to go in ahead of this one that changes behaviour? That way, if we have to bisect to pin down a problem we're going to be looking at a much more focussed change.

Reply:
Ok, first one for the headers to reduce noise in the following ones
https://github.com/elastic/elasticsearch/pull/130800